File: Measurement.cpp
#include "Measurement.hpp"

Measurement::Measurement(const std::string& sensorId, float value, const std::string& timestamp)
	: m_sensorId(sensorId), m_value(value), m_timestamp(timestamp)
{
	std::cout << "Measurement created for sensor: " << sensorId << std::endl;
}

Measurement::~Measurement()
{
	std::cout << "Measurement destroyed: " << m_sensorId << std::endl;
}

bool Measurement::IsAnomalous(float min, float max) const
{
	return (m_value < min || m_value > max);
}

std::string Measurement::ToString() const
{
	return "Sensor " + m_sensorId + " | Value: " + std::to_string(m_value) + " | Time: " + m_timestamp;
}

File: Lab_04/Lab04/Lab04Monitoring/Measurement.cs
// Класс Measurement описывает одно измерение.
// Для демонстрации ЛР-5 добавляем конструктор копирования (через другой объект)
// и аккуратный ToString().
using System;

namespace Lab04Monitoring
{
    public class Measurement
    {
        public string SensorId { get; }
        public float Value { get; }
        public DateTime Timestamp { get; }

        public Measurement(string sensorId, float value, DateTime timestamp)
        {
            SensorId = sensorId;
            Value = value;
            Timestamp = timestamp;
        }

        // "Конструктор копирования" по образцу C++.
        public Measurement(Measurement other)
            : this(other.SensorId, other.Value, other.Timestamp)
        {
        }

        public override string ToString()
        {
            return $"Measurement(Sensor={SensorId}, Value={Value:F2}, Time={Timestamp:O})";
        }
    }
}
